src/Microsoft.ML.OnnxTransform/OnnxTransform.cs

```diff
@@ -80,17 +80,17 @@ public sealed class Arguments : TransformInputBase
             public bool FallbackToCpu = false;
         }
 
-        private readonly Arguments _args;
+        private readonly Options _options;
         internal readonly OnnxModel Model;
 
         internal const string Summary = "Transforms the data using the Onnx model.";
         internal const string UserName = "ONNX Scoring Transform";
         internal const string ShortName = "Onnx";
         internal const string LoaderSignature = "OnnxTransform";
 
-        public readonly string[] Inputs;
-        public readonly string[] Outputs;
-        public readonly ColumnType[] OutputTypes;
+        internal readonly string[] Inputs;
```

Comment:
can these be combined to a tuple?

Reply:
They cannot, the reason is that it's not a `OneToOneTransformer`. So there could be a different number of inputs and outputs.

---
In reply to: [253716288](https://github.com/dotnet/machinelearning/pull/2367#discussion_r253716288) [](ancestors = 253716288)